libcaf_core/src/logger.cpp

```diff
@@ -273,11 +300,15 @@ logger::logger(actor_system& sys) : system_(sys) {
 }
 
 void logger::run() {
-  std::ostringstream fname;
-  fname << "actor_log_" << detail::get_process_id() << "_" << time(0)
-        << "_" << to_string(system_.node())
-        << ".log";
-  std::fstream out(fname.str().c_str(), std::ios::out | std::ios::app);
+  auto filename = system_.config().logger_filename;
+  // Replace node ID placeholder.
+  auto placeholder = std::string{"[NODE]"};
```

Comment:
Creating a heap-stored string here is unnecessary.

```cpp
const char needle[] = "[NODE]";
auto i = std::search(filename.begin(), filename.end(), std::begin(needle), std::end(needle) - 1);
```

-1 because the char array is null-terminated.